server/master/src/main/java/org/apache/accumulo/master/upgrade/Upgrader9to10.java

```diff
@@ -500,4 +506,127 @@ public void upgradeDirColumns(ServerContext ctx, Ample.DataLevel level) {
+          }
+          String metaEntry = key.getColumnQualifier().toString();
+          Path relPath = resolveRelativePath(metaEntry, key);
+          Path absPath = new Path(volumeProp, relPath);
+          if (fs.exists(absPath)) {
+            log.debug("Changing Tablet File path from {} to {}", metaEntry, absPath);
+            Mutation m = new Mutation(key.getRow());
+            // add the new path
+            m.at().family(key.getColumnFamily()).qualifier(absPath.toString())
+                .visibility(key.getColumnVisibility()).timestamp(key.getTimestamp())
+                .put(entry.getValue());
+            // delete the old path
+            m.at().family(key.getColumnFamily()).qualifier(key.getColumnQualifierData().toArray())
+                .visibility(key.getColumnVisibility()).timestamp(key.getTimestamp()).delete();
+            writer.addMutation(m);
+          } else {
+            throw new IllegalArgumentException(
+                "Relative Tablet file " + relPath + " not found at " + absPath);
+          }
+        }
+      }
+    } catch (MutationsRejectedException | TableNotFoundException e) {
+      throw new RuntimeException(e);
+    } catch (IOException ioe) {
+      throw new UncheckedIOException(ioe);
+    }
+  }
+
+  /**
+   * Check if table has any relative paths, return false if none are found. When a relative path is
+   * found, check existence of the file path constructed from the upgrade property + relative path
+   */
+  public static boolean checkForRelativePaths(AccumuloClient client, Configuration hadoopConf,
+      String tableName, String upgradeProperty) {
+    boolean hasRelatives = false;
+
+    try (Scanner scanner = client.createScanner(tableName, Authorizations.EMPTY)) {
+      FileSystem fs = FileSystem.get(hadoopConf);
+      log.info("Looking for relative paths in {}", tableName);
+      scanner.fetchColumnFamily(MetadataSchema.TabletsSection.DataFileColumnFamily.NAME);
+      for (Entry<Key,Value> entry : scanner) {
+        Key key = entry.getKey();
+        Text filePath = key.getColumnQualifier();
+        if (!filePath.toString().contains(":")) {
+          // found relative paths so verify the property used to build the absolute paths
+          hasRelatives = true;
+          if (upgradeProperty == null || upgradeProperty.isBlank()) {
+            throw new IllegalArgumentException(
+                "Missing required property " + Property.INSTANCE_VOLUMES_UPGRADE_RELATIVE.getKey());
+          }
+          Path relPath = resolveRelativePath(key.getColumnQualifier().toString(), key);
+          Path absPath = new Path(upgradeProperty, relPath);
+          if (!fs.exists(absPath)) {
+            throw new IllegalArgumentException("Tablet file " + relPath + " not found at " + absPath
+                + " using volume: " + upgradeProperty);
+          }
+        }
+      }
+    } catch (TableNotFoundException e) {
+      throw new RuntimeException(e);
```

Comment:
`IllegalStateException` is probably better here. Non-specific `RuntimeException` seems unhelpful when communicating what the problem was, and requires callers to catch all possible runtime exceptions if they want to handle this situation... which is a bit unfriendly to the caller and therefore bad practice.

Reply:
I will also use ```IllegalStateException``` in the other method's catch block as well.